servant/src/Servant/API/BoundedNat.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Servant.API.BoundedNat (BoundedNat (unBoundedNat), unsafeBoundedNat) where
+
+import           Control.Monad    (unless)
+import           Data.Aeson
+import           Data.Aeson.Types (modifyFailure)
+import           Data.Bifunctor   (bimap)
+import           Data.Ix
+import           Data.Proxy       (Proxy (Proxy))
+import qualified Data.Text        as T
+import           GHC.Generics     (Generic)
+import           GHC.TypeLits
+import           Servant.API
+
+-- | A newtype wrapper around 'Word' that ensures the value is within a given range.
```

Comment:
Any reason to limit to `Word` when the type-level `Natural` can contain arbitrarily-long integers >=0?